build8.cmd

```diff
@@ -74,96 +67,114 @@
-	set PATH=C:/cygwin64/bin;%PATH%
-
 	rem uncomment to enable AV scan
 	rem set CLAMSCAN=C:/cygwin64/bin/clamscan.exe
 
 	rem local directory to clone JDK repo into
 	set OJDK_SRC_DIR=jdk%OJDK_MILESTONE%%DEV_REPO%
 	set OJDK_SRC_PATH=%WORK_DIR%/%OJDK_SRC_DIR%
 
-	rem the remote repo for JDK sources
-	if defined USE_MERCURIAL (		
-		set OJDK_REMOTE_REPO_PROTOCOL=https://
-		set OJDK_REMOTE_REPO=hg.openjdk.java.net/jdk%OJDK_MILESTONE%/jdk%OJDK_MILESTONE%%DEV_REPO%
-	)
-
 	@echo *** work directory %WORK_DIR%
 	@echo *** JDK tag %OJDK_TAG%
 	@echo *** debug level %OJDK_DEBUG_LEVEL%
 
 	rem ZIP_DIR_NAME is the directory the zipfile (RELEASE_ZIPFILE) will unpack into
-	set ZIP_DIR_NAME=openjdk-%OJDK_MILESTONE%%OJDK_UPDATE%-%OJDK_BUILD%
-	set RELEASE_ZIPFILE=OpenJDK8U-jdk_x64_windows_%OJDK_MILESTONE%%OJDK_UPDATE%%OJDK_BUILD%%EA_SUFFIX%.zip
-	@echo final JDK product will be !RELEASE_ZIPFILE! directory !ZIP_DIR_NAME!
+	if defined BUILD_JDK (
+		set ZIP_DIR_NAME=openjdk-%OJDK_MILESTONE%%OJDK_UPDATE%-%OJDK_BUILD%
+		set RELEASE_ZIPFILE=OpenJDK8U-jdk_x64_windows_%OJDK_MILESTONE%%OJDK_UPDATE%%OJDK_BUILD%%EA_SUFFIX%.zip
+		@echo final JDK product will be !RELEASE_ZIPFILE! directory !ZIP_DIR_NAME!
+	)
 
-	set ZIP_JRE_DIR_NAME=openjdk-%OJDK_MILESTONE%%OJDK_UPDATE%-%OJDK_BUILD%-jre
-	set RELEASE_JRE_ZIPFILE=OpenJDK8U-jre_x64_windows_%OJDK_MILESTONE%%OJDK_UPDATE%%OJDK_BUILD%%EA_SUFFIX%.zip
-	@echo final JRE product will be !RELEASE_JRE_ZIPFILE! directory !ZIP_JRE_DIR_NAME!
+	if defined BUILD_JRE (
+		set ZIP_JRE_DIR_NAME=openjdk-%OJDK_MILESTONE%%OJDK_UPDATE%-%OJDK_BUILD%-jre
+		set RELEASE_JRE_ZIPFILE=OpenJDK8U-jre_x64_windows_%OJDK_MILESTONE%%OJDK_UPDATE%%OJDK_BUILD%%EA_SUFFIX%.zip
+		@echo final JRE product will be !RELEASE_JRE_ZIPFILE! directory !ZIP_JRE_DIR_NAME!
+	)
 
-	call :download_git || exit /b 1
+	time /t
+	call :download_boot_git || exit /b 1
 	set SAVEPATH=%PATH% 
+
 	if defined DOWNLOAD_TOOLS ( 
 		call :download_ojdkbuild || exit /b 1
-		call :download_mercurial || exit /b 1
+	) else (
+		@echo *** skipping JDK tool downloads
+	)
+	
+	call :setsdkenv
+	call :download_git || exit /b 1
+	
+	if defined DOWNLOAD_TOOLS ( 
 		call :build_freetype || exit /b 1
 	) else (
-		@echo *** skipping JDK build downloads
+		@echo *** skipping JDK tool downloads
+		call :download_git || exit /b 1
 	)
+	
+	time /t
 
 	if defined DOWNLOAD_JDK (
 		call :checkout_jdk_source || exit /b 1
 	)
-	
-	if defined CONFIGURE_JDK (
-		call :configure_jdk_build || exit /b 1
-	)
+	time /t
 
 	if defined CLEAN_JDK (
 		call :clean_jdk || exit /b 1
 	)
+	
+	if defined FIX_HEADER_PATCH (
+		@echo *** Add temporary patch to be able to build on Windows
+		@echo on
+		pushd "%WORK_DIR%/%OJDK_SRC_DIR%/hotspot"
+		C:\cygwin64\bin\wget -O JDK-8272214.patch "https://cr.openjdk.java.net/~andrew/openjdk8/8248901-fail/webrev.01/hotspot.patch" || exit /b 1
+		unix2dos JDK-8272214.patch
+		%HG% import --no-commit JDK-8272214.patch || exit /b 1
+		popd
+	)

```

Comment:
We should drop this temp patch code as `JDK-8272214` has been fixed in 8u312 already.

Reply:
Absolutely - I wasn't sure.  Done.